### PR TITLE
doc(querying): replace code examples with graph-client

### PR DESCRIPTION
**Pull request by charlypoly**
@azf20, here are the changes to better showcase our [`graph-client`](https://github.com/graphprotocol/graph-client) in code examples ✨ 

I also changed the content navigation by moving "Querying Best Practices" above "Querying from an Application" because people need to know about GraphQL best practices before talking about client (IMHO)

### CODE REVIEW

File: pages/en/querying/querying-best-practices.mdx [hunk 1]
```diff
@@ -71,9 +71,21 @@ For a complete list of rules with code examples, please look at our GraphQL Vali
...
+
+However, as stated in ["Querying from an Application"](/querying/querying-from-an-application) we recommend you to use our `graph-client` that support unique features such as:
+- Cross-chain Subgraph Handling: Querying from multiple subgraphs in a single query
+- [Automatic Block Tracking](https://github.com/graphprotocol/graph-client/blob/main/packages/block-tracking/README.md)
+- [Automatic Pagination](https://github.com/graphprotocol/graph-client/blob/main/packages/auto-pagination/README.md)
+- Fully typed result
+
+<br />
+
+Here's how to query The Graph with `graph-client`:
```
Comment (charlypoly) [inline pages/en/querying/querying-best-practices.mdx:84]:
```suggestion
It means that you can query a GraphQL API using standard `fetch` (natively or via `@whatwg-node/fetch` or `isomorphic-fetch`).

However, as stated in ["Querying from an Application"](/querying/querying-from-an-application), we recommend you to use our `graph-client` that supports unique features such as:
- Cross-chain Subgraph Handling: Querying from multiple subgraphs in a single query
- [Automatic Block Tracking](https://github.com/graphprotocol/graph-client/blob/main/packages/block-tracking/README.md)
- [Automatic Pagination](https://github.com/graphprotocol/graph-client/blob/main/packages/auto-pagination/README.md)
- Fully typed result

<br />

Here's how to query The Graph with `graph-client`:
```

File: pages/en/querying/querying-best-practices.mdx [hunk 2]
```diff
@@ -84,55 +96,16 @@ query GetToken($id: ID!) {
...
-Available for React, Angular, Vue, Ember, iOS, and Android, Apollo Client, although the heaviest clients, brings many features to build advanced UI on-top of GraphQL:
+main()
+```
 
-- advanced error handling
-- pagination
-- data prefetching
-- optimistic UI
-- local state management
+More GraphQL clients alternatives are covered in ["Querying from an Application"](/querying/querying-from-an-application).
```
Comment (charlypoly) [inline pages/en/querying/querying-best-practices.mdx:108]:
```suggestion
More GraphQL client alternatives are covered in ["Querying from an Application"](/querying/querying-from-an-application).
```

File: pages/en/querying/querying-from-an-application.mdx [hunk 3]
```diff
@@ -22,9 +22,146 @@ Using the GraphQL endpoint, you can use various GraphQL Client libraries to quer
...
+
+The Graph is providing it own GraphQL client, `graph-client` that support unique features such as:
+- Cross-chain Subgraph Handling: Querying from multiple subgraphs in a single query
+- [Automatic Block Tracking](https://github.com/graphprotocol/graph-client/blob/main/packages/block-tracking/README.md)
+- [Automatic Pagination](https://github.com/graphprotocol/graph-client/blob/main/packages/auto-pagination/README.md)
+- Fully typed result
+
+<br />
+
+Also integrated with popular GraphQL clients such as Apollo and URQL and compatible with all environments (React, Angular, Node.js, React Native), using `graph-client` will give you the best experience for interacting with The Graph.
```
Comment (charlypoly) [inline pages/en/querying/querying-from-an-application.mdx:38]:
```suggestion
The Graph is providing it own GraphQL client, `graph-client` that supports unique features such as:
- Cross-chain Subgraph Handling: Querying from multiple subgraphs in a single query
- [Automatic Block Tracking](https://github.com/graphprotocol/graph-client/blob/main/packages/block-tracking/README.md)
- [Automatic Pagination](https://github.com/graphprotocol/graph-client/blob/main/packages/auto-pagination/README.md)
- Fully typed result

<br />

Also integrated with popular GraphQL clients such as Apollo and URQL and compatible with all environments (React, Angular, Node.js, React Native), using `graph-client` will give you the best experience for interacting with The Graph.
```

File: pages/en/querying/querying-from-an-application.mdx [hunk 4]
```diff
@@ -22,9 +22,146 @@ Using the GraphQL endpoint, you can use various GraphQL Client libraries to quer
...
+```
+
+<br />
+
+**⚠️ Important notice**
+
+`graph-client` is perfectly integrated with other GraphQL clients such as Apollo client, URQL or React Query, you will [find examples in the official repository](https://github.com/graphprotocol/graph-client/tree/main/examples).
+
+However, if you choose to go with another client, keep in mind that **you won't be able to get to use Cross-chain Subgraph Handling or Automatic Pagination which are core features for querying The Graph**.
+
```
Comment (charlypoly) [inline pages/en/querying/querying-from-an-application.mdx:149]:
```suggestion
`graph-client` is perfectly integrated with other GraphQL clients such as Apollo client, URQL, or React Query; you will [find examples in the official repository](https://github.com/graphprotocol/graph-client/tree/main/examples).

However, if you choose to go with another client, keep in mind that **you won't be able to get to use Cross-chain Subgraph Handling or Automatic Pagination, which are core features for querying The Graph**.

```

File: pages/en/querying/querying-from-an-application.mdx [hunk 5]
```diff
@@ -22,9 +22,146 @@ Using the GraphQL endpoint, you can use various GraphQL Client libraries to quer
...
+<br />
+
 ### Apollo client
 
-[Apollo client](https://www.apollographql.com/docs/) supports web projects, including frameworks like React and Vue, as well as mobile clients like iOS, Android, and React Native.
+[Apollo client](https://www.apollographql.com/docs/) is the ubiquitous GraphQL client on the front-end ecosystem.
+
+Available for React, Angular, Vue, Ember, iOS, and Android, Apollo Client, although the heaviest clients, brings many features to build advanced UI on-top of GraphQL:
```
Comment (charlypoly) [inline pages/en/querying/querying-from-an-application.mdx:156]:
```suggestion
Available for React, Angular, Vue, Ember, iOS, and Android, Apollo Client, although the heaviest client, brings many features to build advanced UI on top of GraphQL:
```

File: pages/en/querying/querying-from-an-application.mdx [hunk 6]
```diff
@@ -98,9 +235,18 @@ client
...
 ### URQL
 
-Another option is [URQL](https://formidable.com/open-source/urql/), a somewhat lighter-weight GraphQL client library.
+Another option is [URQL](https://formidable.com/open-source/urql/) which is available within Node.js, React/Preact, Vue, Svelte environments, with more advanced features:
+
+- Flexible cache system
+- Extensible design (easing adding new capabilities on top of it)
+- Lightweight bundle (~5x lighter than Apollo Client)
+- Support for file uploads and offline mode
+
```
Comment (charlypoly) [inline pages/en/querying/querying-from-an-application.mdx:248]:
```suggestion
Another option is [URQL](https://formidable.com/open-source/urql/) which is available within Node.js, React/Preact, Vue, and Svelte environments, with more advanced features:

- Flexible cache system
- Extensible design (easing adding new capabilities on top of it)
- Lightweight bundle (~5x lighter than Apollo Client)
- Support for file uploads and offline mode

```

File: pages/en/querying/querying-from-an-application.mdx [hunk 7]
```diff
@@ -22,9 +22,146 @@ Using the GraphQL endpoint, you can use various GraphQL Client libraries to quer
...
+export default App
+```
+
+<br />
+
+**⚠️ Important notice**
+
+`graph-client` is perfectly integrated with other GraphQL clients such as Apollo client, URQL, or React Query; you will [find examples in the official repository](https://github.com/graphprotocol/graph-client/tree/main/examples).
+
+However, if you choose to go with another client, keep in mind that **you won't be able to get to use Cross-chain Subgraph Handling or Automatic Pagination, which are core features for querying The Graph**.
```
Comment (azf20) [inline pages/en/querying/querying-from-an-application.mdx:148]: I would say `querying subgraphs` here rather than The Graph

**Review by azf20 — APPROVED** [review]
This looks good to me!

**Pull request merged**